src/Protobuf/Decode.elm

```diff
@@ -498,6 +500,77 @@ sfixed32 =
     packedDecoder Bit32 (Decode.map (Tuple.pair 4) (Decode.signedInt32 LE))
 
 
+{-| Decode a variable number of bytes into an integer from `-9223372036854775808` to `9223372036854775807`.
+-}
+int64 : Decoder Int64
+int64 =
+    intType Int64.config
+
+
+{-| Decode a variable number of bytes into an integer from `-9223372036854775808` to `9223372036854775807`.
+-}
+sint64 : Decoder Int64
+sint64 =
+    sintType Int64.config
+
+
+{-| Decode a variable number of bytes into an integer from `0` to `18446744073709551615`.
+-}
+uint64 : Decoder Int64
+uint64 =
+    uintType Int64.config
+
+
+{-| Decode a eight bytes into an integer from `0` to `18446744073709551615`.
+-}
+fixed64 : Decoder Int64
+fixed64 =
+    fixed64Type <|
+        Decode.map2 (\lower upper -> Int64.fromInt32s { lower = lower, upper = upper })
+            (Decode.unsignedInt32 LE)
+            (Decode.unsignedInt32 LE)
+
+
+{-| Decode eight bytes into an integer from `-9223372036854775808` to `9223372036854775807`.
+-}
+sfixed64 : Decoder Int64
+sfixed64 =
+    fixed64
+
+
+{-| Decode a VarInt into a data type of your choice.
+-}
+intType : DecodeVarInt intType config -> Decoder intType
+intType config =
+    packedDecoder VarInt <| pack config
+
+
+{-| Decode a VarInt into a data type of your choice.
+Applies the given zagZig function.
+-}
+sintType : DecodeVarInt intType { config | zagZig : intType -> intType } -> Decoder intType
+sintType config =
+    packedDecoder VarInt <| packWith config.zagZig config
+
+
+{-| Decode a VarInt into a data type of your choice.
+Applies the given `toUnsigned` function to make negative results positive.
+-}
+uintType : DecodeVarInt intType { config | toUnsigned : intType -> intType } -> Decoder intType
+uintType config =
+    packedDecoder VarInt <| packWith config.toUnsigned config
+
+
+{-| Decode 64 bits into a data type of your choice.
+If you are using this function make sure that you are always consuming exactly 64 bits.
+-}
+fixed64Type : Decode.Decoder int64Type -> Decoder int64Type
```

Comment:
can we not a have a `fixedDecoder`?

Reply:
I guess we can but we would need to specify how many bytes we want to decode.
For fixed32 it's 4 bytes, but for fixed64 it's 8 bytes contrary to the other (VarInt) decoders.

So I suppose we could add it to the config approach with `{ fixedLength: Int, fixedDecoder: Decode.Decoder intType, fixedEncoder: intType -> Encode.Encoder }`?

Do you like that better than the current approach?

Reply:
Well maybe that's not worth it - now they're both inline which should be ok I guess